src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,165 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
 void testapp::config_controls(OSPMagnum& rOspApp)
 {
     // Configure Controls
+    //Load toml
+    auto data = toml::parse("settings.toml");
+    try 
+    {
+        osp::UserInputHandler& rUserInput = rOspApp.get_input_handler();
+        for (const auto& [k, v] : data.as_table())
+        {
+            std::cout << "Parsing " << k << std::endl;
+            std::string primary = toml::find(v, "primary").as_string();
+            std::vector<osp::ButtonVarConfig> controls = parse_control(primary);
+
+            std::string secondary = toml::find(v, "secondary").as_string();
+            std::vector<osp::ButtonVarConfig> secondaryKeys = parse_control(secondary);
+
+            controls.insert(controls.end(), secondaryKeys.begin(), secondaryKeys.end());
```

Comment:
Here you're failing to use std::move on your osp::ButtonVarConfig objects.

Use
```
controls.insert(controls.end(), std::make_move_iterator(secondaryKeys.begin()), std::make_move_iterator(secondaryKeys.end());
```
to tell the controls variable that it should move from the secondaryKeys array, instead of copy.